docs/pages/release_notes.md

```diff
@@ -14,6 +14,22 @@ This is a {{ site.pmd.release_type }} release.
 
 ### 🚀 New and noteworthy
 
+### 🌟 Rule Changes
+
+Several rules for unit testing have been renamed to better reflect their actual scope. Lots of them were called after JUnit / JUnit 4, even when they applied to JUnit 5 and / or TestNG.
+
+* `java/bestpractices/JUnit4TestShouldUseAfterAnnotation` has been renamed to {% rule java/bestpractices/JUnitTestShouldUseAfterAnnotation %}
+
+* `java/bestpractices/JUnit4TestShouldUseBeforeAnnotation` has been renamed to {% rule java/bestpractices/JUnitTestShouldUseBeforeAnnotation %}
```

Comment:
> Any reason why these two rules are named "JUnitTestShould..." and not just "UnitTestShould..."?
At least, we also have some testng annotation in our xpath expression (not sure, if that makes sense...)

TBH, I hadn't realized the rules include the TestNG annotations… but looking at it, they will never work, both rules still add a `[../MethodDeclaration[pmd-java:hasAnnotation('org.junit.Test')]]` condition, that is bound to JUnit (and JUnit 4 for the matter, so looking at JUnit 5 annotations doesn't really work either even if the doc claimed otherwise).

I'll move this back to draft and rework these rules to properly cover all claimed cases + TestNG. It should be small enough of a change.